### PR TITLE
feat: validate password, alert, responsive site and category filter

### PR DESCRIPTION
**Pull request by PuneetP16**
Added Features:
Alerts,
responsiveness,
scrollToTop,
category selection filter,
Filter Icon for making filter mobile responsive,
signup password validation,

Files to check:
Alert.jsx
responsive.jsx
scrollToTop.jsx

### CODE REVIEW

File: src/components/UI/Alert/Alert.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	const onClickHandler = () => {
+		setAlert((s) => ({
+			...s,
+			visibility: !s.visibility,
+		}));
+	};
+	const { text, visibility, type } = alert;
+
+	useEffect(() => {
+		const timerVar = setTimeout(() => {
```
Comment (m26unkwn) [inline src/components/UI/Alert/Alert.jsx:15]: timerVar can be timer variable right?

File: src/responsive.css [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	}
+
+	.aside--filter {
+		display: none;
+	}
+
+	/* .filter__modal {
+		display: flex;
+	} */
+
```
Comment (m26unkwn) [inline src/responsive.css:81]: remove this comment

**Review by m26unkwn — APPROVED** [review]
Everything is good and clean.

Things that  I like most.

1. Filter icon looking is great.
2. Responsiveness is looking great
3. Code is clean and readable.
4. Rating is looking great.



1.

File: src/components/UI/Alert/Alert.jsx (hunk 1 above)
Comment (PuneetP16) [reply src/components/UI/Alert/Alert.jsx:15]: I didn't got you can you elaborate?

File: src/components/Card/CategoryCard/CategoryCard.jsx [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
-			<Link to="/productlisting" className="categories__links">
+			<Link
+				onClick={() =>
+					dispatch({
+						type: "CATEGORY_SELECTION",
+						payload: prodCat.categoryName,
+					})
+				}
+				to="/productlisting"
+				className="categories__links"
```
Comment (manojsarna) [inline src/components/Card/CategoryCard/CategoryCard.jsx:21]: Suggestion : use Prettier to format the code

File: src/components/Rating/Rating.jsx [hunk 4]
```diff
@@ -2,7 +2,16 @@ import "./Rating.css";
...
+		<div
+			className="card__rating"
+			style={
+				rating > 3
+					? { backgroundColor: "#009000", border: "2px solid #167403" }
+					: {
+							backgroundColor: "var(--COLOR-PRIMARY)",
+					  }
+			}
+		>
```
Comment (manojsarna) [inline src/components/Rating/Rating.jsx:14]: here too

File: src/components/UI/Alert/Alert.jsx (hunk 1 above)
Comment (manojsarna) [inline src/components/UI/Alert/Alert.jsx:15]: Suggestion : it provides the timeoutId , give relevant names easy understanding

File: src/contexts/scrollToTopContext.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import { useFilter } from "./filterContext";
+
+export const ScrollToTopProvider = ({ children }) => {
+	const location = useLocation();
+	const { category, rating, sortByPrice, priceRange } = useFilter();
+
+	useEffect(() => {
+		window.scrollTo(0, 0);
+	}, [location, category, rating, sortByPrice, priceRange]);
+
```
Comment (manojsarna) [inline src/contexts/scrollToTopContext.js:12]: Suggestion : too many dependencies do check

File: src/pages/Authentication/SignUp/SignUp.jsx [hunk 6]
```diff
@@ -36,20 +43,38 @@ export const SignUp = () => {
...
+				loginData,
+				userData,
+				dispatch,
+			});
+		} else {
+			setAlert((a) => ({
+				...a,
+				visibility: true,
+				text: "Minimum 8 char, 1 Uppercase, 1 Lowercase, 1 number & 1 Special Character required",
+				type: "alert--danger",
```
Comment (manojsarna) [inline src/pages/Authentication/SignUp/SignUp.jsx:69]: Alignment check

**Review by manojsarna — APPROVED** [review]
Overall looks good

Do check the comments

File: src/components/Card/CategoryCard/CategoryCard.jsx (hunk 3 above)
Comment (PuneetP16) [reply src/components/Card/CategoryCard/CategoryCard.jsx:21]: i have 4 spaces as default spacing rule, it is formatted.

File: src/components/UI/Alert/Alert.jsx (hunk 1 above)
Comment (PuneetP16) [reply src/components/UI/Alert/Alert.jsx:15]: Yeah nice suggestion. Thanks.

File: src/contexts/scrollToTopContext.js (hunk 5 above)
Comment (PuneetP16) [reply src/contexts/scrollToTopContext.js:12]: this is needed, as i change the state of  any filter it needs to be triggered.

**Pull request merged**